### PR TITLE
add docker publish jobs for server images

### PR DESCRIPTION
**Pull request by TylerWanner**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
adds a circle orb to docker publish prefect server images

## Why is this PR important?
versioned images published to docker hub for open use on each merge to master

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -441,3 +444,35 @@ workflows:
               ignore: /.*/
             tags:
               only: /^[0-9]+\.[0-9]+\.[0-9]+$/
+
+  'Build Earth Images':
```
Comment (cicdw) [inline .circleci/config.yml]:
```suggestion
  'Build Server Images':
```
😛

File: .circleci/config.yml [hunk 2]
```diff
@@ -441,3 +444,35 @@ workflows:
...
+    jobs:
+      - docker/publish:
+          docker-password: DOCKER_HUB_PW
+          docker-username: DOCKER_HUB_USER
+          image: 'prefecthq/apollo'
+          path: 'server/services/apollo'
+          tag: $CIRCLE_SHA1,latest,master
+          filters:
+            branches:
+              only: master
```
Comment (cicdw) [inline .circleci/config.yml]: Can we change this to filter based on tag, similar to the `release_to_pypi` job?  I don't think we need to rebuild these image with every PR, only releases

File: .circleci/config.yml [hunk 3]
```diff
@@ -441,3 +444,35 @@ workflows:
               ignore: /.*/
             tags:
               only: /^[0-9]+\.[0-9]+\.[0-9]+$/
+
+  'Build Earth Images':
+    jobs:
+      - docker/publish:
+          docker-password: DOCKER_HUB_PW
+          docker-username: DOCKER_HUB_USER
```
Comment (cicdw) [inline .circleci/config.yml]: Do these need `$` to be read from the environment?

Comment (TylerWanner) [reply .circleci/config.yml]: @cicdw no actually it's the name of the variable that we're passing to the orb

File: .circleci/config.yml (hunk 2 above)
Comment (TylerWanner) [reply .circleci/config.yml]: @cicdw can we make it master so we can see it run then we can update it to only on releases?

Comment (cicdw) [reply .circleci/config.yml]: Yup for sure, that makes sense

**Pull request merged**